cmd/otel-allocator/target/target.go

```diff
@@ -27,23 +27,25 @@ type LinkJSON struct {
 }
 
 type Item struct {
```

Comment:
I'm a big fan of computing the hash once, but I'm worried that the `Item` struct isn't easy to consume by other packages.  It seems easy to do it wrong:
- someone can create an item, filling out all possible values, and have an empty string for a hash.
- someone can create an item with `NewItem()` and later modify the fields, causing the hash to not match the values.

I'm not sure if it's worth it to change this, some things I thought of:
- don't export `Item` struct - I think this makes things more difficult in this case; this works when a struct is meant to implement an interface.
- exporting `hash` (with the `-` json tag) - then consumers can set their own or use the one provided in `NewItem()`. 
- have `Hash()` check if `hash` is empty, and, if so, calculate the hash and set it (requires a pointer receiver).

I think the easiest solution is just to add a comment documenting that an `Item` shouldn't be made directly and `JobName`, `TargetURL`, and `Labels` shouldn't change.